core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -62,7 +62,8 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
 
   private[spark] def loadFromSystemProperties(silent: Boolean): SparkConf = {
     // Load any spark.* system properties
-    for ((key, value) <- Utils.getSystemProperties if key.startsWith("spark.")) {
+    for ((key, value) <- Utils.getSystemProperties if key.startsWith("spark.")
+      || key.startsWith("pyspark.")) {
```

Comment:
I think you could just call your properties `spark.pyspark.driver.python` and `spark.pyspark.python` then not have to change these checks.  All spark properties should start with that anyway.